notebooks/roc_plot_example.py

```diff
@@ -124,107 +164,109 @@
     verbose=True,
 )
 
+#%% [markdown]
+# Load the *canonical* circuit
+
+d = {(d[0], d[1].hashable_tuple, d[2], d[3].hashable_tuple): False for d in exp.corr.all_edges()}
+d_trues = get_true_edges()
+for k in d_trues:
+    d[k] = True
+exp.load_subgraph(d)
+canonical_circuit_subgraph = deepcopy(exp.corr)
+
 #%% [markdown]
 # <h2> Arthur plays about with loading in graphs </h2>
 # <h3> Not relevant for doing ACDC runs </h3>
 # <p> Get Adria's docstring runs! </p>
 
-from acdc.acdc_utils import false_positive_rate, false_negative_rate, true_positive_rate
-api = wandb.Api()
-runs = api.runs("remix_school-of-rock/acdc")
-filtered_runs = []
-
-for run in tqdm(runs):
-    if "agarriga-docstring" in run.name:
-        filtered_runs.append(run)
+def get_acdc_runs(
+    experiment,
+    project_name: str = ACDC_PROJECT_NAME,
+    run_name_filter: Callable[[str], bool] = ACDC_RUN_FILTER,
+):
+    api = wandb.Api()
+    runs = api.runs(project_name)
+    filtered_runs = []
+    for run in tqdm(runs):
+        if run_name_filter(run.name):
+            filtered_runs.append(run)
+    cnt = 0
+    corrs = []
+    args = project_name.split("/")
+    for run in tqdm(filtered_runs):
+        run_id = run.id
+        try:
+            experiment.load_from_wandb_run(*args, run_id)
+            corrs.append(deepcopy(exp.corr))
+        except Exception as e:
+            print(e)
+            cnt+=1
+            continue
+    return corrs
+
+if "corrs" not in locals(): # this is slow, so run once
+    corrs = get_acdc_runs(exp)
 
-assert len(filtered_runs) > 150, len(filtered_runs)
+#%%
 
-#%% [markdown]
-# <h2> The output log seems malformatted in about 6 cases. I don't know why and it seems OK to just skip these </h2> 
-
-cnt = 0
-corrs = []
-
-for run in tqdm(filtered_runs):
-    run_id = run.id
-    print(len(corrs), "len_corr")
-    try:
-        exp.load_from_wandb_run("remix_school-of-rock", "acdc", run_id)
-        print(exp.count_no_edges())
-        corrs.append(deepcopy(exp.corr))
-    except Exception as e:
-        print(e)
-        cnt+=1
-        continue
+# do SP stuff
 
-# %%
+def get_sp_runs(
```

Comment:
This seems confusing to write: going from a project ID to the subgraph that that SP run found